apps/src/StudioApp.js

```diff
@@ -1323,6 +1323,48 @@ StudioApp.prototype.onMouseMoveVizResizeBar = function (event) {
   this.resizeVisualization(newVizWidth);
 };
 
+StudioApp.prototype.clearVisualizationSize = function () {
+  if ($('#visualizationColumn').hasClass('wireframeShare')) {
+    return;
+  }
+
+  const editorColumn = $(".editor-column");
+  const visualization = document.getElementById('visualization');
+  const visualizationResizeBar = document.getElementById('visualizationResizeBar');
+  const visualizationColumn = document.getElementById('visualizationColumn');
+
+  if (this.isRtl()) {
+    visualizationResizeBar.style.right = '';
+    editorColumn.css('right', '');
+  } else {
+    visualizationResizeBar.style.left = '';
+    editorColumn.css('left', '');
+  }
+
+  visualizationResizeBar.style.lineHeight = '';
+  visualizationColumn.style.maxWidth = '';
+  visualization.style.maxWidth = '';
+  visualization.style.maxHeight = '';
+
+  if (!utils.browserSupportsCssMedia()) {
+    visualization.style.height = '';
+    visualization.style.width = '';
+  }
+
+  applyTransformScaleToChildren(visualization, '');
+
+  $('#soft-buttons').removeClass('soft-buttons-compact');
+
+  const smallFooter = document.querySelector('#page-small-footer .small-footer-base');
+  if (smallFooter) {
+    smallFooter.style.maxWidth = '';
+    const smallPrint = smallFooter.querySelector('small');
+    smallPrint.style.float = '';
+  }
+
+  utils.fireResizeEvent();
```

Comment:
Cool. Those seem like good ideas. I think you're right that an end-to-end eyes test could give us some additional confidence in this.

I do like the idea of continuing to explore if there are ways of accomplishing thumbnail generating without having to mutate our DOM in a bunch of places.

Given that there's still a bit of work to be done here, it might also be worth checking in with Poorva to see if we want to deprioritize thumbnails to get the rest of teacher view finished first.